pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -210,8 +210,9 @@ func (c *Cluster) makeDeployment(name, daemonName string) *extensions.Deployment
 				"prometheus.io/port": strconv.Itoa(metricsPort)},
 		},
 		Spec: v1.PodSpec{
-			Containers:    []v1.Container{c.mgrContainer(name, daemonName)},
-			RestartPolicy: v1.RestartPolicyAlways,
+			ServiceAccountName: "rook-ceph-mgr",
```

Comment:
This would be a good setting to have in the cluster CRD so the admin has control over the service accounts. The osd service account follows this pattern today. 

Previously the osd was the only daemon running with a special SA. The initial thought was that access for other daemons could be added to the same SA and the daemons could use the same SA, thus it was given the generic name `rook-ceph-cluster`. However, with the mgr SA I think it's obvious we need to split this up in to an SA for each daemon, which is already what you've already assumed with this PR. In the cluster CRD I think we would want to specify the service accounts in this manner:
```
  serviceAccounts:
    mgr: rook-ceph-mgr
    osd: rook-ceph-osd
```

To maintain backward compatibility we would need to look at the current `serviceAccount` property to set the osd SA if this new property is not set.

Reply:
I'm thinking that the mgr and OSD are special cases -- in general, other daemon types like RGW, MDS, rbd-mirror, etc should not have special service account properties.

If we hard-code the service account names, the user can still modify the definitions of those service accounts to taste. I wonder if letting them change the names themselves is more customizability than is really needed?

Reply:
I could imagine that for running multiple rook-ceph clusters in a single kubernetes environment, a user might want to have different service accounts for the different clusters. I'm not sure if this makes sense in reality, but it's the only instance I can think of where it would be desired.